src/clr-angular/data/datagrid/datagrid-items.spec.ts

```diff
@@ -45,6 +45,13 @@ export default function(): void {
             expect(this.itemsProvider.displayed).toEqual([1, 2, 3, 4, 5]);
         });
 
+        it("subscribes to changes in the datagrid Items (_items)", function() {
+            // testing the observable subscription in the constructore
+            this.testComponent.numbers.push(6);
+            this.fixture.detectChanges();
+            expect(this.itemsProvider.displayed).toEqual([1, 2, 3, 4, 5, 6]);
```

Comment:
This doesn't test the subscription, you are checking the provider's array here. You could remove the subscription in the directive and this test would still pass.

You simply want to check that the test component now displays the numbers. That's what your new code is doing, giving the responsibility of view creation to the `*clrDgItems` directive.

Reply:
I moved `.datagrid-row-replaced` up to the top level for the export.

Reply:
I updated this test to test the `testContent` of the test element.